test/unit/test-action.js

```diff
@@ -735,6 +746,48 @@ describe('Action method', () => {
     expect(inv.args['key1']).to.equal(11);
     expect(inv.source).to.equal(child);
   });
+
+  describe('macros', () => {
+    let ampActionMacro;
+    let expectedActionInvocation;
+
+    beforeEach(() => {
+      // The expected action invocation for the action macro.
+      expectedActionInvocation = new ActionInvocation(
+          ampActionMacro,
+          'method',
+          {realArgName: 'realArgValue'},
+          ampActionMacro,
+          ampActionMacro,
+          ActionTrust.HIGH,
+          'tap',
+          'AMP-ACTION-MACRO',
+          sinon.match.number
+      );
+      // A caller that references an action macro.
+      targetElement.setAttribute('on', 'tap:action-macro-id.execute(arg1=2)');
+      ampActionMacro = document.createElement('amp-action-macro');
+      ampActionMacro.setAttribute('id', id);
+      ampActionMacro.setAttribute(
+          'action', 'action-macro-id.method(realArgName=arg1)');
```

Comment:
This is an interesting case. What happens if we have a cycle between two or more amp-action-macro?

Reply:
I believe the JS engine should hit a call stack limit. I am not sure we need to guard against that as it is a developer error but we can also just ensure that the caller or triggerNode in the action callback in amp-action-macro is never an amp-action-macro, which would ensure that this cycle wouldn't happen. I also can't think of a use case right now where we would want to do the following:

    button -> amp-action-macro -> amp-action-macro -> amp-component

Though perhaps this might be something that could be of use in the future.

Reply:
The way `amp-bind-macro` deals with this is by only allowing references to macros defined beforehand. See `bind-evaluator.js#addMacros()`.

Reply:
Sounds good. Will do. Thanks Will.